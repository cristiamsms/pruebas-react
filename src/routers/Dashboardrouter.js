import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'
import { DcScreen } from '../components/dc/DcScreen'
import { HeroesScreen } from '../components/heroes/HeroesScreen'
import { MarvelScreen } from '../components/marvel/MarvelScreen'
import { SearchScreen } from '../components/search/SearchScreen'
import { Navbar } from '../components/Ui/Navbar'

export const Dashboardrouter = () => {
    return (
        <>
            <Navbar/>
            <div className="container mt-4 ml-4">
                <Switch>
                    <Route exact path="/marvel" component={MarvelScreen} /> 
                    <Route exact path="/hero/:heroeId" component={HeroesScreen} /> 
                    <Route exact path="/dc" component={DcScreen} />
                    <Route exact path="/Search" component={SearchScreen} />  
                    <Redirect to="/marvel"/> 
                </Switch>
            </div>
        </>
    )
}
