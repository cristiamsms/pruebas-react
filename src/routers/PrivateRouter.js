import React from 'react'
import { Redirect, Route } from 'react-router-dom'
import PropTypes from 'prop-types';

export const PrivateRouter = ({
    isAuthenticated,
    component:Component,
    ...rest
}) => {
    localStorage.setItem('lastPath',rest.location.pathname);
    return (
        <Route {...rest}
        component={(props) =>(
            (  isAuthenticated) 
                ? (<Component {...props}/>)
                : (<Redirect to="/login"/>)
        )

        }>
            
        </Route>
    )
}

PrivateRouter.propTyps = {
    isAuthenticated:PropTypes.bool.isRequired,
    component:PropTypes.func.isRequired
}