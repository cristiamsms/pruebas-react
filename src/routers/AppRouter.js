import React, { useContext } from 'react'

import {
  BrowserRouter as Router,
  Switch
} from "react-router-dom";
import { AuthContext } from '../auth/AuthContext';
import { Login } from '../login/Login';
import { Dashboardrouter } from './Dashboardrouter';
import { PrivateRouter } from './PrivateRouter';
import { PublicRoute } from './PublicRoute';
export const AppRouter = () => {
  const {user} = useContext(AuthContext);
    return (
        <Router>
        <div>
         
  
        
          <Switch>
           <PublicRoute exact path="/login" component={Login} isAuthenticated={user.logged}/>
           <PrivateRouter path="/" component={Dashboardrouter}
                          isAuthenticated={user.logged}/>
          </Switch>
        </div>
      </Router>
    )
}
