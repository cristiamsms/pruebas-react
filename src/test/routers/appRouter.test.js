const { mount } = require("enzyme")
import React from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { AppRouter } from '../../routers/AppRouter';
describe('pruebas en el app router', () => {
    
    const value ={
        dispatch: jest.fn(),
        user:{logged:false}
    }
    
    test('prueba 1', () => {

       const wrapper=mount(
           <AuthContext.Provider value={value}>
  
            <AppRouter/>

           </AuthContext.Provider>
         
       )
        expect(wrapper).toMatchSnapshot();
    });
    test('prueba 2', () => {
        const value ={
            dispatch: jest.fn(),
            user:{logged:true,
            name:'cristian'}
        }
        

        const wrapper=mount(
           <AuthContext.Provider value={value}>
  
            <AppRouter/>

           </AuthContext.Provider>
         
       )
        expect(wrapper.find('.navbar').exists()).toBe(true);
    })
    
    
})
