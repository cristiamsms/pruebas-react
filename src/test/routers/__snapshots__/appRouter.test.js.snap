// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el app router prueba 1 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
